index.js

```diff
@@ -124,7 +124,7 @@ function getOptions(uri, o, method) {
         o._encodingProvided = true;
     }
 
-    o.agentClass = /^https/.test(o.uri) ? httpsAgentClass : httpAgentClass;
+    o.agentClass = /^https/.test(o.uri || o.url) ? httpsAgentClass : httpAgentClass;
```

Comment:
We set up the .uri property further up ([line 73 & following](https://github.com/wikimedia/preq/pull/25/files#diff-168726dbe96b3ce427e7fedce31bb0bcR73)); perhaps it would make sense to add `o.uri = o.uri || o.url;` there, so that it is applied consistently?